versions-common/src/main/java/org/codehaus/mojo/versions/api/PropertyVersions.java

```diff
@@ -59,61 +56,33 @@ public class PropertyVersions extends AbstractVersionDetails {
 
     private final Set<ArtifactAssociation> associations;
 
-    private final VersionsHelper helper;
-
     /**
      * The available versions.
      *
      * @since 1.0-beta-1
      */
-    private final SortedSet<ArtifactVersion> versions;
+    private final SortedSet<ArtifactVersion> resolvedVersions;
+
+    private final Log log;
+
+    private VersionsHelper helper;
 
     private final PropertyVersions.PropertyVersionComparator comparator;
 
-    PropertyVersions(String profileId, String name, VersionsHelper helper, Set<ArtifactAssociation> associations)
-            throws VersionRetrievalException {
+    PropertyVersions(
+            String profileId,
+            String name,
+            Log log,
+            VersionsHelper helper,
+            Set<ArtifactAssociation> associations,
+            SortedSet<ArtifactVersion> resolvedVersions) {
         this.profileId = profileId;
         this.name = name;
+        this.log = log;
```

Comment:
should we check for null value as is in other places ...

Reply:
That's for later. Doing so would make us have to provide this in all the tests and I still wanted to minimize the size of this PR.